### PR TITLE
Add additional engine auto-labelling

### PR DESCRIPTION
**Pull request by cbracken**
This adds additional engine auto-labelling for the following labels:
* embedder
* platform-fuchsia
* platform-linux
* platform-macos
* platform-web
* platform-windows

### CODE REVIEW

**Comment by cbracken** [comment]
Not sure who the right reviewer is but I figure there are 90% odds it's someone on this list. Either way, it might come in handy in the future if it's your first time seeing this code, like it was for me.

**Comment by cbracken** [comment]
@CaseyHillers once this is merged, is this automatically deployed, or is there a manual deploy step we need?

File: app_dart/lib/src/request_handlers/github_webhook.dart [hunk 1]
```diff
@@ -316,12 +316,36 @@ class GithubWebhook extends RequestHandler<Body> {
         hasTests = true;
       }
 
-      if (filename.startsWith('shell/platform/darwin/ios')) {
+      if (filename.startsWith('shell/platform/android')) {
```
Comment (zanderso) [inline app_dart/lib/src/request_handlers/github_webhook.dart:319]: Maybe:
```dart
final List<List<String>> matcher = <List<String>>[
  <String>['shell/platform/android', 'platform-android'],
  <String>['shell/platform/embedder', 'embedder'],
  <String>['shell/platform/darwin/common', 'platform-ios'],
  <String>['shell/platform/darwin/common', 'platform-macos'],
  <String>['shell/platform/darwin/ios', 'platform-ios'],
  <String>['shell/platform/darwin/macos', 'platform-macos'],
  <String>['shell/platform/fuchsia', 'platform-fuchsia'],
  <String>['shell/platform/linux', 'platform-linux'],
  <String>['lib/web_ui', 'platform-web'],
  <String>['web_sdk', 'platform-web'],
];

for (final List<String> matcher in matchers) {
  final String prefix = matcher[0];
  final String label = matcher[1];
  if (filename.startsWith(prefix)) {
    labels.add(label);
  }
}
```

**Review by zanderso — COMMENTED** [review]
Looks like tests are here: https://github.com/flutter/cocoon/blob/master/app_dart/test/request_handlers/github_webhook_test.dart#L412

File: app_dart/lib/src/request_handlers/github_webhook.dart (hunk 1 above)
Comment (cbracken) [reply app_dart/lib/src/request_handlers/github_webhook.dart]: I like it -- or better, a map. Updated.

**Comment by cbracken** [comment]
> Looks like tests are here:

This PR has tests :P Okay, well test. AFAIK all existing tests seem to conflate tests and labelling. Was thinking of creating a separate expectLabels() test utility function for this sort of thing. That feels like an orthogonal patch though so may send that separately.

**Comment by cbracken** [comment]
Actually I'll do that refactor patch first, since it seems useful to test independently.

**Comment by cbracken** [comment]
Done. Just doing it inline in this patch since it's not much code to extract.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Comment by christopherfujino** [comment]
> @CaseyHillers once this is merged, is this automatically deployed, or is there a manual deploy step we need?

there's a daily deploy scheduled.

**Comment by CaseyHillers** [comment]
Lgtm. Thanks for making this a map!

**Pull request merged**